discovery_engine/lib/src/domain/engine/engine.dart

```diff
@@ -88,6 +88,12 @@ abstract class Engine {
     int pageSize,
   );
 
+  /// Performs a deep search regarding a document.
```

Comment:
🎨 it's not clear to me how the arguments relate to the mentioned "document"? is it something along the lines:
Performs a deep search for documents relating to... an existing one described by `term`?
similarly in other places with this docstring

Reply:
good point, the method is meant to be called on document snippets, but it can actually be called on any search term and market, i'll adjust the docs.